form/form.go

```diff
@@ -538,14 +538,15 @@ func (f *Values) Add(key, val string) {
 	f.values = append(f.values, formValue{key, val})
 }
 
-// Encode encodes the values into “URL encoded” form ("bar=baz&foo=quux").
+// Encode encodes the keys and values into “URL encoded” form
+// ("bar=baz&foo=quux").
 func (f *Values) Encode() string {
 	var buf bytes.Buffer
 	for _, v := range f.values {
 		if buf.Len() > 0 {
 			buf.WriteByte('&')
 		}
-		buf.WriteString(url.QueryEscape(v.Key))
+		buf.WriteString(v.Key)
```

Comment:
Switching to this form without an escape worked fine. I thought this might be a breaking change, which is why I initially started with the second encoding method. I think we control the keys tho so this _should_ be fine.

Reply:
Hm, actually now that I think about it, we probably do want to keep the escape — most keys should be safe, but there are a couple free-form areas in the API like `metadata` where we could have arbitrary input passed in.

Would you mind bringing back the `QueryEscape` and adopt the solution I suggested from stripe-ruby (replace encoded square brackets with real square brackets)? I believe that should be pretty safe regardless of use in form body or query string, and allows us to keep the API simple by having only one implementation. Sorry for the churn.

Reply:
Sounds good :)